cli/azd/pkg/workflow/config.go

```diff
@@ -0,0 +1,54 @@
+package workflow
+
+// Stores a map of workflows configured for an azd project
+type WorkflowMap struct {
```

Comment:
Would it have been possible to do `type WorkflowMap map[string]*Workflow` instead?  That would have allowed us to customize the behavior of the YAML marshalling without forcing us to add `Get` and `Set` methods

Reply:
Yes, i did attempt this initially but was running into some issues. I will double back and figure out why.

Reply:
Figured it out on my second pass.  I realized since the type alias is of a map it is implicitly already a pointer and was able to remove some of the complexity of creating new instances, etc which allowed me to simplify.